tpu_inference/tpu_info.py

```diff
@@ -66,13 +66,20 @@ def get_num_cores_per_chip() -> int:
 def get_num_chips() -> int:
     accel_files = glob.glob("/dev/accel*")
     if accel_files:
-        return len(accel_files)
-    try:
-        vfio_entries = os.listdir("/dev/vfio")
-        numeric_entries = [
-            int(entry) for entry in vfio_entries if entry.isdigit()
-        ]
-        return len(numeric_entries)
-    except FileNotFoundError as e:
-        logger.error("Failed to detect number of TPUs: %s", e)
-        return 0
+        num_devices = len(accel_files)
+    else:
+        try:
+            vfio_entries = os.listdir("/dev/vfio")
+            numeric_entries = [
+                int(entry) for entry in vfio_entries if entry.isdigit()
+            ]
+            num_devices = len(numeric_entries)
+        except FileNotFoundError as e:
+            logger.error("Failed to detect number of TPUs: %s", e)
+            return 0
+
+    # For tpu7x, each chip has 2 chiplets exposed as separate devices
+    tpu_type = get_tpu_type()
+    if tpu_type and "tpu7x" in tpu_type.lower():
+        return num_devices // 2
```

Comment:
not really a fan of hard coding the value "2" or having "tpu7x" specific logic.

Is there any logic within tpu_info that programatically fetches num_devices?

Or wouldn't a simple `len(jax.devices())` return correct number of cores - regardless of having 1 chiplet or 1 chiplets?